### ISSUE TITLE
BatchNorm shouldn't use Bessel's correction for a batch size of 1

### ISSUE THREAD

**Issue by apaszke** [pytorch/pytorch#1381]
Right now it makes everything `NaN`.

**Comment by colesbury** [comment]
To be clear, for a `N x C x H x W` tensor, the problem is only when all of N, H, and W are 1. (So BatchNorm2d on batch size of 1 is OK as long as you don't have a 1x1 image).

What's the desired behavior? The only reasonable behavior I can think of is:

- Raise an exception when the dimensions of which you are normalizing are one
- Output zero (+ optional affine transform)
